main.py

```diff
@@ -1,21 +1,52 @@
-from bs4 import BeautifulSoup
 from datetime import datetime
-from termcolor import colored
+
 import requests
+from bs4 import BeautifulSoup
+from termcolor import colored
+
 
-while True:
-    name = input('Find out when famous people were born. Enter a name or type q to quit: ')
-    if name != 'q':
-        try:
-            url = requests.get('https://www.wikipedia.org/wiki/' + name)
-            soup = BeautifulSoup(url.content, features='html.parser')
-            table = soup.find('table', class_="infobox")
-            date = table.find('span', class_="bday")
-            date_format = datetime.strptime(date.contents[0], '%Y-%m-%d').strftime('%d %B %Y')
-            print(colored('\n{} was born on {}\n'.format(name, date_format), 'blue'))
-        except AttributeError:
-            print(colored("\nI don't know this person. Try again.\n", 'red'))
+def ask_for_input():
+    user_input = input(
+        "Find out when famous people were born. Enter a name or type q to quit: "
+    )
+    return user_input
+
+
+def show_error_message(response):
+    if response.status_code == 404:
+        print(colored("\nI don't know this person. Try again.\n", "red"))
+    elif response.status_code == 504:
+        print(colored("\nThe system is down.\n", "red"))
     else:
-        print('\nBye!')
-        break
+        print(colored("\nUnknown error.\n", "red"))
+
+
+def main():
+    while True:
+        user_input = ask_for_input()
+        if user_input != "q":
+            response = requests.get("https://www.wikipedia.org/wiki/" + user_input)
+            if response.status_code == 200:
+                soup = BeautifulSoup(response.content, features="html.parser")
+                try:
+                    table = soup.find("table", class_="infobox")
+                    date_element = table.find("span", class_="bday")
+                    formatted_date = datetime.strptime(
+                        str(date_element.contents[0]), "%Y-%m-%d"
```

Comment:
**issue:** Here error handing should be improved, if span is empty `date_element.contents[0]` will return `IndexError`. Also if string is not in required format `%Y-%m-%d` `datetime.strptime` will return `ValueError`.